client/src/pages/Programmes/useProgrammes.js

```diff
@@ -4,11 +4,18 @@ import CoursePlannerClient from "../../common/CoursePlannerClient";
 const useProgrammes = () => {
     const [programmeDegrees, setProgrammeDegrees] = useState([]);
 
-    const createProgramme = (programme, updateList) => {
-        CoursePlannerClient.createProgramme(programme)
+    const createProgramme = async (programme, updateList) => {
+        let newProgramme;
+
+        await CoursePlannerClient.createProgramme(programme)
+            .then((res) => {
+                newProgramme = res;
+            })
             .then(() => updateList())
             .then(() => fetchAllProgrammes())
             .catch((e) => console.error(e));
+
+        return newProgramme;
```

Comment:
Since this is an `async` function maybe we can make this a bit cleaaner by doing this?

```
const createProgramme = async (programme, updateList) => {
    try {
       const newProgramme = await CoursePlannerClient.createProgramme(programme);
       updateList(); // if updateList is async then put a await here too
       fetchAllProgrammes();
    } catch (e) {
        console.log(e)
    }


}
```

Reply:
Thanks, done!